test/parallel_exec.jl

```diff
@@ -801,7 +805,7 @@ end
 @test [1:100...] == pmap(x->x, Base.Generator(x->(sleep(0.0001); x), 1:100))
 
 # Test asyncmap
-@test allunique(asyncmap(x->object_id(current_task()), 1:100))
+@test allunique(asyncmap(x->(sleep(1.0);object_id(current_task())), 1:10))
```

Comment:
Previously each item was being executed in a new task. Now the same set of tasks run through the entire collection. The `sleep` is required as otherwise the first task itself will map the entire range (as the computation is non-yielding)